kolibri/plugins/facility/assets/src/routes.js

```diff
@@ -24,21 +24,45 @@ import {
 } from './modules/classAssignMembers/handlers';
 import { PageNames } from './constants';
 
-function facilityParamRequiredGuard(toRoute, subtopicName) {
-  if (store.getters.userIsMultiFacilityAdmin && !toRoute.params.facility_id) {
-    router.replace({
-      name: 'ALL_FACILITIES_PAGE',
-      params: { subtopicName },
-    });
-    return true;
+function facilityParamRequiredGuard(toRoute, fromRoute, subtopicName) {
+  if (
+    store.getters.userIsMultiFacilityAdmin &&
+    !toRoute.params.facility_id &&
+    fromRoute.query.subtopicName !== subtopicName
+  ) {
+    const currentSubtopicName = toRoute.query.subtopicName;
+    if (currentSubtopicName !== subtopicName) {
```

Comment:
The `toRoute` will never have a subtopicName because that's only a defined parameter for the ALL_FACILITIES page, which we are not using this guard on (and definitely shouldn't).